source/LumiCalReco/src/LumiCalClusterer_getCalHits.cpp

```diff
@@ -78,7 +78,9 @@ int LumiCalClustererClass::getCalHits(	EVENT::LCEvent * evt,
 
       } else {
 	arm =(*_mydecoder)( calHitIn )["barrel"]; // from 1 and 2
-	phiCell = (*_mydecoder)( calHitIn )["phi"];
+	//FIXME: add check for phi range?
+	phiCell = (*_mydecoder)( calHitIn )["phi"]; // goes from -phiMax/2 to +phiMax/2
+	if(phiCell < 0 ) phiCell += _cellPhiMax; // need to put into positive range only
```

Comment:
It fixes one of the problems, namely that the value in the internal cellID encoding gives values of ~1000 and then the calculation of the phi position is wrong